lib/opencensus/trace.ex

```diff
@@ -1,25 +1,31 @@
 defmodule Opencensus.Trace do
-  @doc "Wraps the given block in a tracing child span with the given label/name and optional attributes"
+  @doc """
+  Wraps the given block in a tracing child span with the given label/name
+  and optional attributes
+  """
   defmacro with_child_span(label, attributes \\ quote(do: %{}), do: block) do
     line = __CALLER__.line
     module = __CALLER__.module
     file = __CALLER__.file
     function = format_function(__CALLER__.function)
 
-    computed_attributes =
-      compute_attributes(attributes, %{
-        line: line,
-        module: module,
-        file: file,
-        function: function
-      })
-
     quote do
       parent_span_ctx = :ocp.current_span_ctx()
 
+      attributes =
+        Map.merge(
+          %{
+            line: unquote(line),
+            module: unquote(module),
+            file: unquote(file),
+            function: unquote(function)
+          },
+          unquote(attributes)
+        )
+
       new_span_ctx =
         :oc_trace.start_span(unquote(label), parent_span_ctx, %{
-          :attributes => unquote(computed_attributes)
+          attributes: :maps.filter(fn _k, v -> v != nil end, attributes)
```

Comment:
This change probably should be moved to the core, so user could remove values that are contained in current context as well.